src/Orleans.Transactions/State/TransactionalState.cs

```diff
@@ -257,7 +272,11 @@ Task ITransactionalResource.Abort(long transactionId)
                 records.ForEach(kvp => this.log.Remove(kvp.Key));
             }
 
-            await Persist(transactionId, this.writeLowerBound);
+            this.metadata.StableVersion = stableversion;
+            this.metadata.HighestVersion = this.version;
+            this.metadata.HighestReadTransactionId = this.highestReadTransactionId;
```

Comment:
Q: is it safe to mutate metadata in several methods?

Reply:
It is only modified in persistence related calls, which are only performed in serial execution by the serial executor.